BillTechPaymentsUpdater.php

```diff
@@ -106,8 +106,12 @@ function updatePayments()
 				continue;
 			}
 
-			$ids = $DB->GetCol("SELECT id FROM billtech_payments WHERE token=? or reference_number=?", array($payment->token, $payment->reference_number));
-			if (!$ids || !count($ids)) {
+			$ids = $DB->GetCol("SELECT id FROM billtech_payments WHERE reference_number=?", array($payment->reference_number));
+			if ($ids && count($ids) > 1) {
+				$ids = $DB->GetCol("SELECT id FROM billtech_payments WHERE token=?", array($payment->token));
```

Comment:
To chyba nie pomoże, to raczej w liście płatności zwróconej z API trzeba sprawdzić czy jest więcej niż jedna płatność z danym reference_number. W tej chwili dla pierwszej płatności zadziała, a dla kolejnej zostanie zwrócony jeden wiersz z billtech_payments i dla drugiej płatności nie zadziała.